src/main/java/com/emc/mongoose/storage/driver/coop/netty/http/s3/S3StorageDriver.java

```diff
@@ -89,9 +105,44 @@ public S3StorageDriver(
 					final int batchSize)
 					throws IllegalConfigurationException, InterruptedException {
 		super(stepId, itemDataInput, storageConfig, verifyFlag, batchSize);
-		final var httpConfig = storageConfig.configVal("net-http");
-		fsAccess = httpConfig.boolVal("fsAccess");
-		versioning = httpConfig.boolVal("versioning");
+		final var objectConfig = storageConfig.configVal("object");
+		fsAccess = objectConfig.boolVal("fsAccess");
+		final var taggingConfig = objectConfig.configVal("tagging");
+		taggingEnabled = taggingConfig.boolVal("enabled");
+		if(taggingEnabled) {
+			Loggers.MSG.info("{}: S3 Object Tagging Mode Enabled", stepId);
+		}
+		final var tags = taggingConfig.<String>mapVal("tags");
+		final StringBuilder exprBuff = new StringBuilder();
+		exprBuff.append(TAGGING_HEADER);
+		for(final var tagEntry: tags.entrySet()) {
+			final var k = tagEntry.getKey();
+			final var v = tagEntry.getValue();
+			exprBuff
+				.append(TAGGING_ENTRY_START)
+				.append(k)
+				.append(TAGGING_ENTRY_MIDDLE)
+				.append(v)
+				.append(TAGGING_ENTRY_END);
+		}
+		exprBuff.append(TAGGING_FOOTER);
+		final var taggingContentExpr = exprBuff.toString();
+		if(taggingEnabled) {
+			Loggers.MSG.debug("{}: tagging content pattern: {}", stepId, taggingContentExpr);
+		}
+		if(
```

Comment:
Why, in the case of asynchronous (and other) markers, do you use `CompositeExpressionInput`, and in other `ConstantValueInput`?

Reply:
The constant string input may be used ff there's no any expression marker (not an expression actually, just a plain string)